Rust unit test build failure

This error occurs in the Rust build of PR #40 which makes unrelated go changes.  It seems the Rust dependencies have suffered bitrot over time and needs some TLC from someone know knows the situation here.

```
error: failed to parse manifest at `/github/home/.cargo/registry/src/index.crates.io-6f17d22bba15001f/redb-2.6.0/Cargo.toml`
Caused by:
  feature `edition2024` is required
  The package requires the Cargo feature called `edition2024`, but that feature is not stabilized in this version of Cargo (1.83.0-nightly (eaee77dc1 2024-09-19)).
  Consider trying a more recent nightly release.
  See https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#edition-2024 for more information about the status of this feature.
Error: Process completed with exit code 101.
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -32,8 +32,8 @@ jobs:
         run: |
           curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
           . "$HOME/.cargo/env"
-          rustup toolchain install nightly-2024-09-28
-          rustup default nightly-2024-09-28
+          rustup toolchain install nightly-2025-06-20
+          rustup default nightly-2025-06-20
       - name: Run Rust tests
         run: |
           . "$HOME/.cargo/env"

--- a/pex/src/lib.rs
+++ b/pex/src/lib.rs
@@ -1,3 +1,6 @@
+#![feature(generic_const_exprs)]
+#![allow(incomplete_features)]
+
 mod macros;
 mod pex_constants;
 use constants::{L, M, N, NS, VL};

--- a/pex/src/main.rs
+++ b/pex/src/main.rs
@@ -1,3 +1,6 @@
+#![feature(generic_const_exprs)]
+#![allow(incomplete_features)]
+
 use std::sync::{Arc, Mutex};
 
 use colored::*;

